File: backend/Model/issueModel.js

const mongoose = require('mongoose')


const Schema = mongoose.Schema
const issueSchema = new Schema({
    
    driver_id: {
        type: Schema.Types.ObjectId,
        required: true,
    },

    manager_id: {
        type: Schema.Types.ObjectId, 
        required: true
    },

    driver_name: {
        type: String, 
        required: true
    },

    content: {
        type: String,
        required: true
    },
    status: {
        type: String,
        default: "pending"
    },
    response: {
        type: String,
        default: ""
    }
    
}, {timestamps: true})

module.exports = mongoose.model('Issue', issueSchema)

File: flutter_app/lib/issue/models/issueModel.dart
class Issue {
  String id;
  String driverId;
  String managerId;
  String driverName;
  final String content;
  String status;
  String response;

  Issue({
    this.id = '',
    this.driverId = '',
    this.managerId = '',
    this.driverName = '',
    required this.content,
    this.response = '',
    this.status = "pending",
  });

  factory Issue.fromJson(Map<String, dynamic> json) {
    return Issue(
      id: json['_id'],
      driverId: json['driver_id'],
      driverName: json['driver_name'],
      managerId: json['manager_id'],
      response: json['response'],
      content: json['content'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    Map<String, dynamic> json = {};
    json["_id"] = id;
    json["driver_id"] = driverId;
    json["driver_name"] = driverName;
    json["manager_id"] = managerId;
    json["response"] = response;
    json["content"] = content;
    json["status"] = status;
    return json;
  }
}
